net/xia/ppal_ether/main.c

```diff
@@ -309,10 +341,12 @@ static int main_newroute(struct xip_ppal_ctx *ctx, struct fib_xid_table *xtbl,
 		rc = -EINVAL;
 		goto free_mem;
 	}
+	xia_ether_header_cache(mether, lladdr);
```

Comment:
Since variable `lladdr` is going away (see review of patch **xia/ether: add main table**), you can replace it with `&id[sizeof(*p)]`.